com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/ProtoBuilder.java

```diff
@@ -65,35 +72,82 @@ static public GeneratedMessageV3.Builder<?> newBuilder(String ext) throws Compil
         }
     }
 
-    protected B transform(Task<Void> task, IResource resource, B messageBuilder) throws IOException, CompileExceptionError {
+    protected B transform(Task task, IResource resource, B messageBuilder) throws IOException, CompileExceptionError {
         return messageBuilder;
     }
 
+    /**
+     * Scan proto message and create a sub-task for each resource in it
+     * @param builder message or builder of the file that should be scanned
+     * @param taskBuilder the builder where result should be applied to
+     */
+    protected void createSubTasks(MessageOrBuilder builder, Task.TaskBuilder taskBuilder) throws CompileExceptionError {
```

Comment:
this function moved from Builder because it's only Proto-resources related approach